### PR TITLE
fix: handle invalid json-rpc request

### PR DESCRIPTION
**Pull request by sergey-melnychuk**
Fixes https://github.com/eqlabs/pathfinder/issues/992

### CODE REVIEW

**Review by Mirko-von-Leipzig — APPROVED** [review]
LGTM - did you check for any other unwraps lurking about?

**Review by kkovaacs — APPROVED** [review]

File: crates/rpc/src/versioning.rs [hunk 1]
```diff
@@ -67,11 +67,11 @@ pub async fn prefix_rpc_method_names_with_version(
     };
 
     let body = if is_single {
-        let mut request: jsonrpsee::types::Request<'_> = serde_json::from_slice(&body).unwrap();
+        let mut request: jsonrpsee::types::Request<'_> = serde_json::from_slice(&body)?;
```
Comment (CHr15F0x) [inline crates/rpc/src/versioning.rs:70]: Is the reply from the server consistent with v0.4.5?

Comment (CHr15F0x) [reply crates/rpc/src/versioning.rs:70]:
```
v0.4.5

Request:
{"jsonrpc":"2.0","id":"4","ethod":"starknet_getBlockWithTxs","params":["latest"]}
Reply:
{"jsonrpc":"2.0","error":{"code":-32600,"message":"Invalid request"},"id":"4"}

Request:
{"jsonrpc":"2.0","idx":"4","method":"starknet_getBlockWithTxs","params":["latest"]}
Reply:
{"jsonrpc":"2.0","error":{"code":-32700,"message":"Parse error"},"id":null}

```

After the fix there's no reply when those errors are encountered.

**Comment by sergey-melnychuk** [comment]
Now (after the force push) error is returned: `{"jsonrpc":"2.0","error":{"code":-32700,"message":"Parse error"},"id":null}`

**Comment by CHr15F0x** [comment]
> Now (after the force push) error is returned: `{"jsonrpc":"2.0","error":{"code":-32700,"message":"Parse error"},"id":null}`

But you won't get the other case right, ie when id is valid there should be `Invalid request` instead.

**Comment by sergey-melnychuk** [comment]
The CI looks funny, but I am sure it has nothing to do with the change.

**Pull request merged**